src/common/coding/encodeTree.js

```diff
@@ -16,14 +16,15 @@ function encode(value, { version, isGraph } = {}) {
   function pushLink($ref, $ver, props, $val, $chi) {
     const [range, _] = splitRef($ref);
 
+    const node = !isEmpty(props)
+      ? makeNode(range ? [{ $key: range, ...props }] : props, undefined, $ver)
+      : isDef($chi)
+      ? makeNode(range ? [{ $key: range, $chi }] : $chi, undefined, $ver)
+      : null;
+
     // prettier-ignore
-    let children =
-      !isEmpty(props) ? makeNode(
-        range ? [{ $key: range, ...props}] : props, undefined, $ver
-      ).children :
-      isDef($chi) ? makeNode(
-        range ? [{ $key: range, $chi }] : $chi, undefined, $ver
-      ).children :
```

Comment:
Previously it was assumed that if props wasn't empty, makeNode will return something.

But if props is something like `{ foo: {} }`, makeNode (correctly) returns undefined.

When this happens on a node with $ref, this code path is taken.